src/core/solver.py

```
@@ -16,17 +15,31 @@
+
+    Args:
+        solver_inst (cp_model.CpSolver): The OR-Tools solver instance to configure.
+    """
+    solver_inst.parameters.linearization_level = 0
+    solver_inst.parameters.symmetry_level = 2
 
 
 class SolutionPrinter(cp_model.CpSolverSolutionCallback):
     """Callback to log intermediate solutions found by the solver."""
 
     def __init__(self, start_time: float):
```

Ruff (0.15.12)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)